src/HealthGPS/kevin_hall_model.cpp

```diff
@@ -195,53 +253,110 @@ void KevinHallModel::set_energy_intake(Person &person) const {
     }
 }
 
-SimulatePersonState KevinHallModel::simulate_person(Person &person, double shift) const {
-    // Initial simulated person state.
-    const double H_0 = person.get_risk_factor_value("Height"_id);
-    const double BW_0 = person.get_risk_factor_value("Weight"_id);
-    const double PAL_0 = person.get_risk_factor_value("PhysicalActivity"_id);
-    const double F_0 = person.get_risk_factor_value("BodyFat"_id);
-    const double L_0 = person.get_risk_factor_value("LeanTissue"_id);
-    const double ECF_0 = person.get_risk_factor_value("ExtracellularFluid"_id);
-    const double G_0 = person.get_risk_factor_value("Glycogen"_id);
-    const double EI_0 = person.get_risk_factor_value("EnergyIntake"_id);
-    const double CI_0 = person.get_risk_factor_value("Carbohydrate"_id);
+void KevinHallModel::initialise_kevin_hall_state(Person &person,
+                                                 std::optional<double> adjustment) const {
 
-    // TODO: Compute height.
-    double H = H_0;
+    // Apply optional weight adjustment.
+    if (adjustment.has_value()) {
+        person.risk_factors.at("Weight"_id) += adjustment.value();
+    }
+
+    // Get already computed values.
+    double H = 0; // TODO: person.risk_factors.at("Height"_id);
+    double BW = person.risk_factors.at("Weight"_id);
+    double PAL = person.risk_factors.at("PhysicalActivity"_id);
+    double EI = person.risk_factors.at("EnergyIntake"_id);
+
+    // Body fat.
+    double F;
+    if (person.gender == core::Gender::female) {
+        F = BW * (0.14 * person.age + 39.96 * log(BW / pow(H / 100, 2.0)) - 102.01) / 100.0;
+    } else if (person.gender == core::Gender::male) {
+        F = BW * (0.14 * person.age + 37.31 * log(BW / pow(H / 100, 2.0)) - 103.94) / 100.0;
```

Comment:
Personally, I'd put these two formulas in their own functions using simple double arguments, making them easy to test, and describe in docstring where the formulas are coming from (paper, reference, etc.)

Reply:
Problem is I have no idea what these are. I think this should be part of a larger reorganisation and labelling effort down the line, when we have time to get them to go through it and name everything.

Reply:
#268